Src/IronPython/Compiler/Ast/PythonNameBinder.cs

```diff
@@ -250,6 +250,7 @@ internal class PythonNameBinder : PythonWalker {
 
         public override bool Walk(ListComprehension node) {
             node.Parent = _currentScope;
+            node.Scope.Parent = _currentScope;
```

Comment:
Why is `ListComprehension` different from `SetComprehension` and `DictionaryComprehension`?

Reply:
Because I forgot to clean it up. Thanks for the catch.